File: tests/talos/java/fact/src/main/java/com/talostrading/Factorial.java
package com.talostrading;

public class Factorial {

    /**
     * Compute n factorial
     * @param n offset in the sequence, should be greater than or equal to zero
     * @return factorial of n
     */
    public static int factorial(final int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}

File: tests/talos/dotnet/fact/fact/Factorial.cs
namespace fact
{
    internal class Fact
    {
        /// <summary>
        ///     Compute n factorial
        /// </summary>
        /// <param name="n">Offset in the sequence, should be greater than or equal to zero</param>
        /// <returns>Factorial of n</returns>
        public static int Factorial(int n)
        {
            if (n == 0) return 1;
            return n * Factorial(n - 1);
        }
    }
}